rfd/XXXX-tsh-upgrade.md

```diff
@@ -0,0 +1,117 @@
+The default installation method will be downloading the pre-built binaries.
+
+#### Linux
+
+Teleport currently maintains DEB and RPM repositores.
+So before upgrading, the upgrade command needs to identify whether the system it's running on is a Debian or Red Hat distribution.
+The upgrade command will look at the /etc/lsb-release or /etc/redhat-release files to identify the Linux distribution and to determine which package manager to use.
+
+The installation method can be further narrowed down by using the package manager CLI tools dpkg for Debian, and rpm for Red Hat.
+If Teleport was installed using the manager, dpkg or rpm can be used to identify which release was installed: teleport, teleport-ent, or teleport-ent-fips.
+Once the release has been identified, the Teleport client tools can now be upgraded using the package manager.
+
+If Teleport was not installed using the package manager, the Teleport client tools will be upgraded by downloading the pre-built binaries.
+
+#### macOS
+
+On macOS, the Teleport package is available for installation using the macOS package installer.
+To detect whether or not Teleport was installed using the installer, the pkgutil command can be used to search for the Teleport package.
+Only Teleport Open Source and Teleport Enterprise are supported on macOS.
+
+A signed release of tsh is also supported on macOS and is required to use TouchID.
+So tsh will need to be upgraded separately after the initial upgrade of the other Teleport client tools.
+
+#### Windows
+
+Only the tsh command is supported on Windows machines.
+tsh only supports an Open Source release, so for windows machines, the tsh upgrade command simply needs to download the desired tsh binary and install it.
+
+## UX
+
+The tsh upgrade command should simply upgrade the Teleport client tools while providing some basic observability about the currently installed version of the Teleport client tools, the Teleport cluster version, and what version the client tools will be upgraded to.
+The upgrade command will require root privileges.
+
+If the Teleport client tools are up to date, simply output the client and server version and exit
+```console
+$ sudo tsh upgrade
+Teleport v13.1.2
+Proxy version 13.1.2
+Teleport client tools are already up to date. Nothing to be upgraded.
+```
+
+If the Teleport client tools are behind and need to be upgraded, the user will be prompted to upgrade
+```console
+$ sudo tsh upgrade
+Teleport v12.0.0
+Proxy version v13.1.2
+Teleport client tools version is behind the Teleport cluster version, do you wish to upgrade? [y/N]:
+no
+Exiting without upgrading Teleport client tools. Teleport servers are compatible with clients that are on the same major version or one major version older. We recommend updating before Teleport client tools become incompatible. If you would like to upgrade manually, installation instructions can be found at https://goteleport.com/docs/installation/
```

Comment:
If I'm on a Debian server A, which is running `teleport` (an agent, or a full-featured Teleport cluster) and want to login to a Teleport instance running on a server B (let's say cloud). Cloud is running a different version than server A. What happens when I run `sudo tsh update`? Does it update the whole teleport package, including my `teleport` process running?